File: supabase/functions/fastapi/scripts/setup_minio.js
/* eslint-disable no-console */
const { Client } = require('minio');

async function main() {
  const endpointVar = process.env.S3_ENDPOINT || 'http://localhost:9000';
  const endpoint = endpointVar.replace('http://', '').replace('https://', '');
  const useSSL = endpointVar.startsWith('https://');
  const accessKey = process.env.S3_ACCESS_KEY || 'minio';
  const secretKey = process.env.S3_SECRET_KEY || 'minio12345';
  const bucket = process.env.S3_BUCKET || 'fiber-photos';

  const minioClient = new Client({ endPoint: endpoint.split(':')[0], port: Number(endpoint.split(':')[1] || (useSSL ? 443 : 9000)), useSSL, accessKey, secretKey });

  const exists = await minioClient.bucketExists(bucket).catch(() => false);
  if (!exists) {
    await minioClient.makeBucket(bucket).catch((e) => console.error('makeBucket error', e));
  }
  console.log(`Bucket ${bucket} is ready`);
}

main().catch((e) => {
  console.error(e);
  process.exit(1);
});

File: supabase/functions/fastapi/scripts/setup_minio.py
import os
from minio import Minio
from minio.lifecycleconfig import LifecycleConfig, Rule, Expiration, Filter


def main():
    endpoint = os.getenv("S3_ENDPOINT", "http://localhost:9000").replace("http://", "").replace("https://", "")
    access_key = os.getenv("S3_ACCESS_KEY", "minio")
    secret_key = os.getenv("S3_SECRET_KEY", "minio12345")
    bucket = os.getenv("S3_BUCKET", "fiber-photos")

    client = Minio(endpoint, access_key=access_key, secret_key=secret_key, secure=endpoint.startswith("https"))

    if not client.bucket_exists(bucket):
        client.make_bucket(bucket)

    # Add per-tenant prefixes lifecycle to expire tmp objects after 30d
    rules = [
        Rule(
            rule_id="tmp-expire",
            status="Enabled",
            expiration=Expiration(days=30),
            filter=Filter(prefix="tmp/")
        )
    ]
    lc = LifecycleConfig(rules=rules)
    client.set_bucket_lifecycle(bucket, lc)
    print(f"Configured bucket {bucket} with lifecycle rules")


if __name__ == "__main__":
    main()

